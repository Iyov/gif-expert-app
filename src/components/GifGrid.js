import React from 'react'; //, { useState, useEffect }
import { useFetchGifts } from '../hooks/useFetchGifts';
//import { GifGridItem } from './GifGridItem';
//import { getGifts } from '../helpers/getGifts';

export const GifGrid = ({ category }) => {
    
    //const [images, setImages] = useState([]);
    const { data, loading } = useFetchGifts( category );

    console.log(data, loading)

    return (
        <>
            <h3>{ category }</h3>

            { loading ? 'Cargando...' : 'Data cargada' }

            {/*<div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img } 
                        />
                    ))
                    
                }
            </div>*/}
        </>
    )
}
