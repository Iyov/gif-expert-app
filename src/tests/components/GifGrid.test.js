import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifts } from '../../hooks/useFetchGifts';
jest.mock('../../hooks/useFetchGifts');

describe('Pruebas en <GifGrid />', () => {
    
    const category = 'Prueba';
    
    test('Debe de retornar el componente correctamente', () => {
        
        useFetchGifts.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifts', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/test1.jpg',
            title: 'Prueba 1'
        },{
            id: '123',
            url: 'https://localhost/test2.jpg',
            title: 'Prueba 2'
        }];

        useFetchGifts.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
    
});
