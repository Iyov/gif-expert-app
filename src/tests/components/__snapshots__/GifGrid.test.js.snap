// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifts 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Prueba
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Prueba 1"
      url="https://localhost/test1.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Prueba 2"
      url="https://localhost/test2.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de retornar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Prueba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
